packages/block-editor/src/components/off-canvas-editor/appender.js

```diff
@@ -27,18 +31,70 @@ export const Appender = forwardRef( ( props, ref ) => {
 		};
 	}, [] );
 
+	const { insertedBlockAttributes } = useSelect(
+		( select ) => {
+			const { getBlockAttributes } = select( blockEditorStore );
+
+			return {
+				insertedBlockAttributes: getBlockAttributes( insertedBlock ),
+			};
+		},
+		[ insertedBlock ]
+	);
+
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const setAttributes =
+		( insertedBlockClientId ) => ( _updatedAttributes ) => {
+			updateBlockAttributes( insertedBlockClientId, _updatedAttributes );
+		};
```

Comment:
We provide a customised implementation of `setAttributes` to the `updateAttributes` method. This is because in this context we do not have access to the block's `setAttributes` method. Therefore we must dispatch attribute updates directly to the store.

The function is curried in order that we can partially apply the blockID in advance.